FlowCrypt/src/androidTest/java/com/flowcrypt/email/rules/RetryRule.kt

```diff
@@ -33,11 +35,21 @@ class RetryRule(private val retryCount: Int = 0) : BaseRule() {
           .firstOrNull()
           ?.value
 
-        val attempts = if (fromAnnotation == null || fromAnnotation !in 1..MAX_RETRY_VALUE) {
+        val hasFlakyAnnotation = Objects.nonNull(
+          description
+            .annotations
+            .filterIsInstance<FlakyTest>()
+            .firstOrNull()
+        )
+
+        //if a test is FlakyTest we will use 2*attempts
+        val value = (if (fromAnnotation == null || fromAnnotation !in 1..MAX_RETRY_VALUE) {
           if (retryCount in 1..MAX_RETRY_VALUE) retryCount else 1
         } else {
           fromAnnotation
-        }
+        }) * if (hasFlakyAnnotation) 2 else 1
+
+        val attempts = if (value in 1..MAX_RETRY_VALUE) value else 1
```

Comment:
`val attempts = value.coerceIn(1..MAX_RETRY_VALUE)`
Here, if a value will be greater than `MAX_RETRY_VALUE` we will use `MAX_RETRY_VALUE`. For now, `MAX_RETRY_VALUE = 100`.

But I prefer to use a min value if somebody will use a big value like 200 instead of 20(like a mistake with more zero). Instead of running a test 100 times, we will run it 1 time. It will be easier to debug it on my side.

Reply:
Aha I see, that makes sense.
Then we can leave above logic as it is now.